src/amr/data/field/refine/magnetic_refine_patch_strategy.hpp

```diff
@@ -0,0 +1,425 @@
+#ifndef PHARE_AMR_MAGNETIC_REFINE_PATCH_STRATEGY_HPP
+#define PHARE_AMR_MAGNETIC_REFINE_PATCH_STRATEGY_HPP
+
+#include "SAMRAI/hier/PatchLevel.h"
+#include "amr/resources_manager/amr_utils.hpp"
+#include "SAMRAI/xfer/RefinePatchStrategy.h"
+#include "core/utilities/constants.hpp"
+#include "core/utilities/index/index.hpp"
+
+#include <cassert>
+
+namespace PHARE::amr
+{
+using core::dirX;
+using core::dirY;
+using core::dirZ;
+
+template<typename ResMan, typename FieldDataT>
+class MagneticRefinePatchStrategy : public SAMRAI::xfer::RefinePatchStrategy
+{
+public:
+    using Geometry        = typename FieldDataT::Geometry;
+    using gridlayout_type = typename FieldDataT::gridlayout_type;
+
+    static constexpr std::size_t dimension = FieldDataT::dimension;
+
+    MagneticRefinePatchStrategy(ResMan& resourcesManager)
+        : rm_{resourcesManager}
+    {
+    }
+
+    void registerIDs(std::optional<int> bx_id, std::optional<int> by_id, std::optional<int> bz_id)
+    {
+        if (bx_id)
```

Comment:
not sure why they're optionals, is this resource manager related?

Reply:
I feel this is a left-over from the second method when we used the oldLevel and sometimes didn't need it because refining not from regrid?

It seems we need these variables because once in postProcess SAMRAI only gives us the coarse and fine patches but we are left to know with which variable concretely to work.

But I find this way of doing this a bit too low level.
As we have discussed, I think, I would rather take VecFields in the MagneticRefinePatchStrategy, they are views anyway, and since we have the RM, we can set them on the coarse and fine patches as needed.

Reply:
it does, but if you can be sure that the resource should always exist in a particular context you can dreference it to drop the optional part

see https://github.com/PhilipDeegan/PHARE/blob/patchghostless/src/amr/resources_manager/resources_manager.hpp#L323-L331